packages/cypress/integration/demo_reflect.ts

```diff
@@ -0,0 +1,89 @@
+function addCard(column, text) {
+  cy.get(`[data-cy=phase-item-editor-${column}-wrapper]`)
+      .as('column-phase-item-editor')
+    
+  cy.get('@column-phase-item-editor')
+    .type(`${text}`)
+    .should('have.text', `${text}`)
+
+  cy.get('@column-phase-item-editor')
+    .type('{enter}')
+  
+  cy.wait(500)
+
+}
+
+function editCard(column, oldText, newText) {
+  cy.get(`[data-cy=reflection-stack-${column}-card-0-wrapper]`)
+    .as('reflection-card-edit')
+
+  cy.get('@reflection-card-edit')
+    .should('have.text', `${oldText}`)
+
+  cy.get('@reflection-card-edit')
+    .type('{selectall}').type('{backspace}').type(`${newText}`)
+    .should('have.text', `${newText}`)
+
+}
+
+function deleteCard(column) {
+      
+  cy.get(`[data-cy=reflection-stack-${column}-card-0-delete]`)
+    .as('reflection-card-delete')
+
+  cy.get('@reflection-card-delete').click()
+}
+
+describe('Test Reflect page Demo', () => {
+
+  before(function() {
+    // runs before each test in the block
+    cy.visit('/retrospective-demo/reflect')
+    cy.get('[data-cy=start-demo-button]').should('be.visible').click()
+
+    cy.get('[data-cy=sidebar-header]').find('button').should('be.visible').click()
+
+  })
+
+  it('Test help menu toggle', () => {
+
+    cy.get('[data-cy=help-menu-toggle]').should('be.visible')
+    
+    cy.get('[data-cy=help-menu-close]').should('be.visible').click()
+
+    cy.get('[data-cy=help-menu-close]').should('not.exist')
+    
+    cy.get('[data-cy=help-menu-toggle]').should('be.visible').click().get('[data-cy=help-menu-close]').should('be.visible')
+
+  })
+
+  it('Test adding, editing, and deleting reflections', () => {
+
+    addCard('Start', 'Start column reflection')
+
+    addCard('Stop', 'Stop column reflection')
+
+    addCard('Continue', 'Continue column reflection')
+
+    editCard('Start', 'Start column reflection', 'Edit reflection')
+
+    editCard('Stop', 'Stop column reflection', 'Edit reflection')
+
+    editCard('Continue', 'Continue column reflection', 'Edit reflection')
+  
+    deleteCard('Start')
+
+    deleteCard('Stop')
+
+    deleteCard('Continue')
+  })
+
+  it('Test if can advance to groups', () => {
+    cy.wait(5000)
```

Comment:
any way we can shorten this? when we get a lot of tests going it'll be nice to keep the suite as fast as possible

Reply:
Looks like I can bring it down to 2000. But for some reason, if I go lower than that, it won't register the click half the time. 

I initially put that in because I noticed that the advancement button will be disabled until a certain period of time. I'd like to not have to use an arbitrary waiting time, but I can't figure out how to wait until a button is ready to be clicked. Do you have an idea on how to do that?

Reply:
hmmm
cypress has a built-in interval retry mechanism for selectors if it doesn't find them immediately.
maybe write a selector like `button:not([disabled])')`? that way it'll keep looking for a button that's not disabled & when it finds it, then it'll click it.

Reply:
Huh, looks like it's working now, with or without the :not(:disabled). I don't even need the wait anymore. I'm didn't change anything, so I'm not sure what happened.

Reply:
Just found this article, which pretty much states and seems to solve my problem: 
https://www.cypress.io/blog/2019/01/22/when-can-the-test-click/

Reply:
Do you mind If I install that cypress-pipe package for this? It seems to do the trick.